tests/evaluation_test.py

```diff
@@ -50,47 +87,55 @@ def test_simple_evaluation(dirty=False, force=False):
     assert 1 == len(results_display_mock.print_cache_file.call_args_list)
 
 
-def test_grouped_evaluation():
-    label = [i % 2 for i in range(100)]
-    dataset = DataFrame({
-        'a': label,
-        'b': label,
-        'g': [i // 2 for i in range(100)]
-    })
+@pytest.mark.parametrize("dirty, force", [(False, False), (False, True), (True, False), (True, True)])
+def test_grouped_evaluation(dirty, force, grouped_dataset):
 
     cross_validation = GroupShuffleSplit(n_splits=1, test_size=0.5)
     predictor = DummyClassifier(strategy='constant', constant=1)
     scorer = make_scorer(accuracy_score)
-    experiment = Experiment(dataset, 'b', predictor, cross_validation, scorer, group_column='g')
+
+    experiment = Experiment(grouped_dataset, 'b', predictor, cross_validation, scorer, group_column='g')
 
     git_mock = Mock()
-    git_mock.is_dirty.return_value = False
+    git_mock.is_dirty.return_value = dirty
+    git_mock.head.object.hexsha = MagicMock()
     results_repo_mock = Mock()
     results_display_mock = Mock()
     runner = ExperimentRunner(git_mock, results_repo_mock, results_display_mock)
 
-    runner.run_experiment(experiment, "", False)
+    try:
+        runner.run_experiment(experiment, "msg", force)
+    except DirtyRepoError:
+        if dirty is True and force is False:
+            # Expected behaviour.
+            return
+        else:
+            raise
 
     assert 1 == len(results_repo_mock.save_results.call_args_list)
     run_info, dataset_info = results_repo_mock.save_results.call_args[0]
 
     print("Run infos", run_info)
 
     expected_results = {'test_score': {'accuracy_score': 0.5}, 'test_score_sem': {'accuracy_score': 0.0}}
-    assert expected_results == run_info['results']
+    pruned_results = {
+        'test_score': {
+            'accuracy_score': run_info['results']["test_score"]["accuracy_score"]
+        },
+        'test_score_sem': {
+            'accuracy_score': run_info['results']["test_score_sem"]["accuracy_score"]
+        }
+    }
```

Comment:
This is a bit weird, why not just test the individual bits of the run info for the values you expect?